packages/ui/src/components/icon-v2/icon-v2.tsx

```diff
@@ -23,15 +23,31 @@ const iconVariants = cva('cn-icon', {
   }
 })
 
-export interface IconPropsV2 extends SVGProps<SVGSVGElement> {
-  name: IconV2NamesType
+interface BaseIconPropsV2 extends SVGProps<SVGSVGElement> {
   size?: VariantProps<typeof iconVariants>['size']
   // incase size will be added through CSS
   skipSize?: boolean
 }
 
-const IconV2: FC<IconPropsV2> = ({ name, size = 'sm', className, skipSize = false }) => {
-  const Component = IconNameMapV2[name]
+interface IconDefaultPropsV2 extends BaseIconPropsV2 {
+  name: IconV2NamesType
+  fallback?: never
+}
+
+interface IconFallbackPropsV2 extends BaseIconPropsV2 {
```

Comment:
Lets remove this type and update `IconDefaultPropsV2` type as below.

```ts
  fallback?: IconV2NamesType
```

Purpose of `fallback` is not only to identify absence of `name`, but also wrong icon name from dynamic sources like API calls.